src/mimefactory.rs

```diff
@@ -1314,6 +1325,17 @@ impl<'a> MimeFactory<'a> {
             let json = self.msg.param.get(Param::Arg).unwrap_or_default();
             parts.push(context.build_status_update_part(json));
         } else if self.msg.viewtype == Viewtype::Webxdc {
+            let topic = create_random_topic();
+            let peer = Box::pin(context.get_or_generate_iroh_keypair())
+                .await?
+                .public();
+            context
+                .iroh_add_peer_for_topic(self.msg.id, topic, peer, None)
+                .await?;
+            headers.protected.push(Header::new(
```

Comment:
Is it guaranteed that the message is encrypted? `IrohGossipTopic` is a secret afaiu

Reply:
I don't think we need to worry about encryption here.  If we want to prevent sending an Iroh-announcement via unencrypted mails we should  do it more generically and not spawn/instantiate anything.  For now, i think it's fine to allow announces via unencrypted mails.

Reply:
It is also put into the protected headers (part of the encrypted part) so should be fine, no? @link2xt